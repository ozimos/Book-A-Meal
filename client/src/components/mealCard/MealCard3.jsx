import React from 'react';
import PropTypes from 'prop-types';
import '../../../public/styles/book_a_meal.css';
import '../../../public/styles/auth.scss';
import '../../../public/styles/modalOpenButton.scss';

const MealCard3 = ({ id, title, imageUrl, price, description, ...props }) => {
  const meal = { id, title, imageUrl, price, description };
  return (
    <React.Fragment>
      <div className="card">
        <img src={imageUrl} alt="Meal" className="fluid-img" />
        <div className="title-element flexbox wrap">
          <div className="shrink">
            {title}
          </div>
          <div className="flexbox">

            {props.addToOrder &&
              <button className="btn title-button icon" onClick={() => props.addToOrder(meal)}>
                &#10004;
              </button>}
            {props.removeFromOrder &&
              <button className="btn title-button icon" onClick={() => props.removeFromOrder(id)}>
                &#10006;
              </button>}
          </div>

        </div>
      </div>
    </React.Fragment>
  );
};

MealCard3.defaultProps = {
  addToOrder: null,
  removeFromOrder: null,
};
MealCard3.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  addToOrder: PropTypes.func,
  removeFromOrder: PropTypes.func,
};
export { MealCard3 };
export default MealCard3;

