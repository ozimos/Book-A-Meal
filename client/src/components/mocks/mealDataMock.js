export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
