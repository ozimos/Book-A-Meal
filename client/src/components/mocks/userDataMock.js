export const userData = {
  data: { id: '123', email: 'asdf' },
  message: 'abc',
  token: 'abc'
};
export const userInput = {
  data: { email: 'asdf' }
};
export const userIncludes = {
  email: 'asdf',
  firstName: 'first',
  lastName: 'last'
};
