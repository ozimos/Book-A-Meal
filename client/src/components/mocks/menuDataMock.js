import { allMeals } from './mealDataMock';

export default {
  data: { Meals: allMeals.data }
};