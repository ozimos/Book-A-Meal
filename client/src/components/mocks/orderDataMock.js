import { allMeals } from './mealDataMock';

const Meals = allMeals.data.map(meal => ({ ...meal, MealOrders: { quantity: 2 } }));
export const order = {
  data: { id: 'orderId',
    userId: 'userId',
    Meals,
    mealList: ['abc', 'def'],
    quantityList: [2, 2] }
};
export const allOrders = {
  data: [{ ...order.data, id: 'order1' }, { ...order.data, id: 'order2' }]
};