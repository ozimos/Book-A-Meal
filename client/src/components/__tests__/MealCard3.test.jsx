/* global React:false, shallow:false toJson:false */

import MealCard3 from '../mealCard/MealCard3';
import { meal } from '../mocks/mealDataMock';

describe('MealCard3 Component', () => {
  it('should render without throwing an error', () => {
    expect(shallow(<MealCard3
      {...meal.data}
      dispatch={jest.fn()}
      addToOrder={jest.fn()}
      removeFromOrder={jest.fn()}
    />)
      .exists(<div className="card" />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = shallow(<MealCard3
      {...meal.data}
      dispatch={jest.fn()}
      addToOrder={jest.fn()}
      removeFromOrder={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

