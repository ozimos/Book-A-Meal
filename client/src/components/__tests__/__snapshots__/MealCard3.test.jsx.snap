// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealCard3 Component renders correctly 1`] = `
<React.Fragment>
  <div
    className="card card-booking pillar"
  >
    <img
      alt="Meal"
      className="fluid-img"
      src="nice"
    />
    <div
      className="pillar"
    >
      <div
        className="title-element text_left"
      >
        nice
      </div>
      <div
        className="flexbox"
      >
        <div
          className="text_left pad"
        >
          ₦
          200
        </div>
        <button
          className="btn title-button icon"
          onClick={[Function]}
        >
          ✔
        </button>
        <button
          className="btn title-button icon"
          onClick={[Function]}
        >
          ✖
        </button>
      </div>
    </div>
  </div>
</React.Fragment>
`;
